### PR TITLE
[SPARK-23662][SQL] Support selective tests in SQLQueryTestSuite

### PR DESCRIPTION
**Pull request by maropu**
## What changes were proposed in this pull request?
This pr supported selective tests in `SQLQueryTestSuite`, e.g.,
```
SPARK_SQL_QUERY_TEST_FILTER=limit.sql,random.sql build/sbt "sql/test-only *SQLQueryTestSuite"
```

## How was this patch tested?
Manually checked

### CODE REVIEW

**Comment by maropu** [comment]
cc: @gatorsmile

**Comment by maropu** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 1]
```diff
@@ -51,6 +51,11 @@ import org.apache.spark.sql.types.StructType
  *   SPARK_GENERATE_GOLDEN_FILES=1 build/sbt "sql/test-only *SQLQueryTestSuite"
  * }}}
  *
+ * For selective tests, run:
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Why not directly using something like 
> build/sbt "~sql/test-only *SQLQueryTestSuite -- -z inline-table.sql -z random.sql"

Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: oh, I didn't know `sbt` can do. I usually use `maven` and it seems `maven` cannot do that. Anyway, it's okay to use `sbt for this case. Thanks!

**Pull request closed without merging**